lib/services/role.go

```diff
@@ -118,41 +132,330 @@ type Access interface {
 	DeleteRole(name string) error
 }
 
+const (
+	// ForwardAgent is SSH agent forwarding.
+	ForwardAgent = "ForwardAgent"
+)
+
+const (
+	// Allow is the set of conditions that allow access.
+	Allow RoleConditionType = "allow"
+	// Deny is the set of conditions that prevent access.
+	Deny RoleConditionType = "deny"
+)
+
+// RoleConditionType specifies the type: allow or deny.
+type RoleConditionType string
+
+// Check checks if RoleConditionType has a valid value.
+func (c RoleConditionType) Check() error {
```

Comment:
Isn't this re-implementation of a `bool` type? The role format has just two values: allow and deny (it's not extensible) so having a simple `type RoleConditionType bool` would do the same thing (even better, because it defaults by itself into `deny` vs an empty string in your case)